src/COMPONENTS/WELCOMEPAGE/WelcomePage.tsx

```diff
@@ -0,0 +1,78 @@
+import { useEffect } from "react";
+import { Link, useNavigate } from "react-router-dom";
+import { useSelector, useDispatch } from "react-redux";
+// @ts-ignore
+import { setProfileDetails } from "../../REDUX/profileDetailSlice.js";
+
+import "./welcomepage.css";
+export const WelcomePage = () => {
+  const navigate = useNavigate();
+  const dispatch = useDispatch();
+  const { name, id, adminRights } = useSelector(
+    (state: any) => state.profileDetailSlice
+  );
+
+  const displayName = name || "Page";
+
+  useEffect(() => {
+    // @ts-ignore
+    const isLoggedIn = JSON.parse(localStorage.getItem("isLoggedIn"));
```

Comment:
Hello @sudhirwork - in my opinion, the `welcome` page should not be aware of how/where the state of `isLoggedIn` is.  That should be behind a utility method, to hide the implementation details.